# Admittance parameters for mass damper systems (mobile robot, platftorm, arm)
# with params (mass_robot, damping_robot), (mass_platform, damping_platform),
# and (mass_arm, damping_arm) respectively

p_pi: &p_pi 3.14159265358979323846
p_2pi: &p_2pi 6.283185307179586232 
p_pi_2: &p_pi_2 1.57079632679489661923
p_2pi_3: &p_2pi_3 2.094395102393195263
p_pi_10: &p_pi_10 0.3141592653589793116 
p_pi_3: &p_pi_3 1.047197551
m_pi: &m_pi -3.14159265358979323846
m_2pi: &m_2pi -6.283185307179586232 
m_pi_2: &m_pi_2 -1.57079632679489661923
m_2pi_3: &m_2pi_3 -2.094395102393195263
m_pi_10: &m_pi_10 -0.3141592653589793116 

mass_1: &mass_1 4
mass_2: &mass_2 2
damp_1: &damp_1 20
damp_2: &damp_2 10

#  SIMULATION
vel_limits: [1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0]
#vel_limits: [0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5]
#vel_limits: [*p_2pi_3,*p_2pi_3,*p_pi,*p_pi,*p_pi,*p_pi, 1.0, 1.5]

acc_limits: [0.2, 0.2, 0.2, 0.2, 0.2, 0.2, 0.2, 0.2]
#acc_limits: [0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 1.5, 2.5]
#acc_limits: [0.7, 0.7, 0.7, 0.7, 0.7, 0.7, 0.7, 1.5, 2.5]
#acc_limits: [*p_pi_10,*p_pi_10,*p_pi_10,*p_pi_10,*p_pi_10,*p_pi_10,*p_pi_10,*p_pi_10]


# REAL ROBOT
#vel_limits: [*p_2pi_3,*p_2pi_3,*p_pi,*p_pi,*p_pi,*p_pi, 1.0, 1.5]
#acc_limits: [*p_pi_10,*p_pi_10,*p_pi_10,*p_pi_10,*p_pi_10,*p_pi_10, 5.0, 6.0]

mass_robot: [*mass_1,0,0,0,0,0,
             0,*mass_1,0,0,0,0,
             0,0,*mass_1,0,0,0,
             0,0,0,*mass_2,0,0,
             0,0,0,0,*mass_2,0,
             0,0,0,0,0,*mass_2]

damping_robot: [*damp_1,0,0,0,0,0,
                0,*damp_1,0,0,0,0,
                0,0,*damp_1,0,0,0,
                0,0,0,*damp_2,0,0,
                0,0,0,0,*damp_2,0,
                0,0,0,0,0,*damp_2]
                
mass_platform: [3,0,0,0,0,0,
                0,3,0,0,0,0,
                0,0,3,0,0,0,
                0,0,0,1,0,0,
                0,0,0,0,1,0,
                0,0,0,0,0,1]

damping_platform: [30,0,0,0,0,0,
                   0,30,0,0,0,0,
                   0,0,30,0,0,0,
                   0,0,0,15,0,0,
                   0,0,0,0,15,0,
                   0,0,0,0,0,15]

mass_arm: [6,0,0,0,0,0,
           0,6,0,0,0,0,
           0,0,6,0,0,0,
           0,0,0,3,0,0,
           0,0,0,0,3,0,
           0,0,0,0,0,3]

damping_arm: [30,0,0,0,0,0,
              0,30,0,0,0,0,
              0,0,30,0,0,0,
              0,0,0,15,0,0,
              0,0,0,0,15,0,
              0,0,0,0,0,15]


mass_platform_prova: [0.1,0,
                      0,0.1]

damping_platform_prova:  [0.2,0,
                          0,0.2]


mass_arm_prova: [10,0,0,0,0,0,
                 0,10,0,0,0,0,
                 0,0,10,0,0,0,
                 0,0,0,10,0,0,
                 0,0,0,0,10,0,
                 0,0,0,0,0,10]

damping_arm_prova: [90,0,0,0,0,0,
                    0,90,0,0,0,0,
                    0,0,90,0,0,0,
                    0,0,0,90,0,0,
                    0,0,0,0,90,0,
                    0,0,0,0,0,90]


mass_robot_prova: [10,0,0,0,0,0,0,0,
                   0,10,0,0,0,0,0,0,
                   0,0,10,0,0,0,0,0,
                   0,0,0,10,0,0,0,0,
                   0,0,0,0,10,0,0,0,
                   0,0,0,0,0,10,0,0,
                   0,0,0,0,0,0,0.1,0,
                   0,0,0,0,0,0,0,0.1]

damping_robot_prova: [90,0,0,0,0,0,0,0,
                      0,90,0,0,0,0,0,0,
                      0,0,90,0,0,0,0,0,
                      0,0,0,90,0,0,0,0,
                      0,0,0,0,90,0,0,0,
                      0,0,0,0,0,90,0,0,
                      0,0,0,0,0,0,0.2,0,
                      0,0,0,0,0,0,0,0.2]
